### ISSUE TITLE
Goals section

### ISSUE THREAD

**Issue by jonathanrobie** [biblicalhumanities/treedown#17]
We should have a high level goals section.  Here's my first shot - let's try to come up with a list we agree on.

*Expressivity*

- Maintains sentence order, while accurately describing constituent structure
- Supports analysis at various levels of granularity
- Clearly distinguishes adjuncts from arguments
- Good for showing the syntax of long and complex sentences
- Minimal overhead for showing the syntax of simple sentences
- Formally defined, with well defined equivalence to bracket notation and Lowfat XML format
- Extensible, allowing custom functionality to be added
- Based on a linguistic model easy to explain to students learning from traditional grammars

*Usability*

- Easy to write using widely available tools
- Easy to read in sentence order
- Good figure / ground characteristics - draws attention to the text
- Easy to exchange in email, issues lists, and forums
- Supports comments

**Comment by jtauber** [comment]
I wonder if "maintain sentence order" can be formally specified as: "if all Treedown labels and annotations are removed, one is left with the original words of the text, in their original order" or similar.

**Comment by jonathanrobie** [comment]
> I wonder if "maintain sentence order" can be formally specified as: "if all Treedown labels and annotations are removed, one is left with the original words of the text, in their original order" or similar.

That's a good way to explain it.

**Comment by christopherland** [comment]
If the goal is "Easy to exchange in email, issues lists, and forums," then we also need to consider text input on mobile devices. In particular, this may have some bearing on the tabs vs. spaces issue.

**Comment by jonathanrobie** [comment]
Some comments from today's meeting:

**James Tauber** 
One thing I’d like to clarify is that “Based on a linguistic model easy to explain to students learning from traditional grammars” doesn’t mean we resort to traditional analyses and ignore a century of descriptive linguistic work but just that we use broadly agreed upon descriptive terms, avoid framework-specific jargon, etc

**Jonathan Robie** 
Ideally, I would like it to be usable by someone who (1) learned via Smyth / Robertson, (2) learned via Mounce, (3) learned via Porter, (4) learned via Runge / Fresch/ Black / Buth

**James Tauber** 
it’s one reason I’ve been looking stuff up in CGEL, NOT because an analysis of English is necessary relevant to an analysis of Greek but because they tend to be modern and precise without being jargon-heavy or framework-specific

**Christopher Land** 
The main categories are fairly generic at this point, even if different models view them differently or label them differently

**Micheal Palmer**
And where we need to refine a term, we can do that as long as we document what we mean and make it available to all.

**Comment by jonathanrobie** [comment]
We discussed granularity of representation with respect to goals, and agreed that the ability to represent the same text at different levels of granularity is an explicit goal for both doing the analysis and consuming it.  A person who is marking down a text might start coarse, then add more detail in successive passes.  A person who is reading a text that is fully marked down in the system might start by seeing a coarse breakdown, then drill down to specific parts of the sentence for more detail.  

For instance, here is a coarse representation:

```
s πολλοὶ 
v ἐπεχείρησαν 
od ἀνατάξασθαι διήγησιν περὶ τῶν πεπληροφορημένων ἐν ἡμῖν πραγμάτων,
```

The same text in the next level of analysis:

```
s πολλοὶ 
v ἐπεχείρησαν 
od
        v.inf ἀνατάξασθαι 
        od διήγησιν 
        + περὶ τῶν πεπληροφορημένων ἐν ἡμῖν πραγμάτων,
```

And in the next level of analysis:

```
s πολλοὶ 
v ἐπεχείρησαν 
od
        v.inf ἀνατάξασθαι 
        od διήγησιν 
        + περὶ 
            τῶν 
            + v πεπληροφορημένων  (: Embedded clause - using adjunct :)
              + ἐν ἡμῖν  
            πραγμάτων,

**Comment by jtauber** [comment]
One goal, not explicitly mentioned here but I think mentioned in conversation, is ability to embed treedown in markdown documents (and have markdown formatters that support extensions be able to do something custom with that treedown)

**Comment by jonathanrobie** [comment]
> One goal, not explicitly mentioned here but I think mentioned in conversation, is ability to embed treedown in markdown documents (and have markdown formatters that support extensions be able to do something custom with that treedown)

Yes, I have been doing that in some of the examples I have been sharing, and I think it is important.
